src/base/bittorrent/sessionimpl.cpp

```diff
@@ -2569,52 +2569,20 @@ bool SessionImpl::cancelDownloadMetadata(const TorrentID &id)
 
 void SessionImpl::increaseTorrentsQueuePos(const QList<TorrentID> &ids)
 {
-    using ElementType = std::pair<int, const TorrentImpl *>;
-    std::priority_queue<ElementType
-        , std::vector<ElementType>
-        , std::greater<ElementType>> torrentQueue;
-
-    // Sort torrents by queue position
-    for (const TorrentID &id : ids)
-    {
-        const TorrentImpl *torrent = m_torrents.value(id);
-        if (!torrent) continue;
-        if (const int position = torrent->queuePosition(); position >= 0)
-            torrentQueue.emplace(position, torrent);
-    }
-
     // Increase torrents queue position (starting with the one in the highest queue position)
-    while (!torrentQueue.empty())
-    {
-        const TorrentImpl *torrent = torrentQueue.top().second;
+    for (TorrentImpl *torrent : asConst(getQueuedTorrentsByID(ids)))
```

Comment:
```suggestion
    for (const TorrentImpl *torrent : asConst(getQueuedTorrentsByID(ids)))
```

Reply:
I would prefer not to do this, as it involves modifying the torrent (albeit indirectly, so C++ technically allows it to be marked as `const`).